avr-hal-generic/src/spi.rs

```diff
@@ -75,9 +75,29 @@ macro_rules! impl_spi {
                 sclk: $sclkmod:ident::$SCLK:ident,
                 mosi: $mosimod:ident::$MOSI:ident,
                 miso: $misomod:ident::$MISO:ident,
+                cs: $csmod:ident::$CS:ident,
             }
         }
     ) => {
+
+        /// Wrapper for the CS pin
+        ///
+        /// Used to contain the chip-select pin during operation to prevent its mode from being
+        /// changed from Output.  Implements
+        /// [`OutputPin`](https://docs.rs/embedded-hal/latest/embedded_hal/digital/v2/trait.OutputPin.html)
+        /// so that it can still be controlled.
```

Comment:
I'd write something like:

_[...] prevent its mode from being changed from Output.  This is necessary because the SPI state machine would otherwise reset itself to SPI slave mode immediately._

_This wrapper can be used just like an output pin, because it implements all the same traits from embedded-hal._